npm/test/errorhandling.test.ts

```diff
@@ -0,0 +1,84 @@
+import { handleUploadError } from "../src/errorhandling"
+
+beforeEach(() => {
+  jest.resetAllMocks()
+  jest.resetModules()
+})
+
+test("handleUploadError function displays error message and throws error with additional information", () => {
+  setupErrorHtml()
+
+  const mockErrorMessage: string = "some error"
+  const mockError = new Error(mockErrorMessage)
+
+  expect(() => {
+    handleUploadError(mockError, "Some additional information")
+  }).toThrowError(new Error("Some additional information: some error"))
+
+  checkExpectedErrorHtmlState(mockErrorMessage)
+})
+
+test("handleUploadError function displays error message and throws error without additional information", () => {
+  setupErrorHtml()
+
+  const mockErrorMessage: string = "some error"
+  const mockError = new Error(mockErrorMessage)
+
+  expect(() => {
+    handleUploadError(mockError)
+  }).toThrowError(new Error("Upload failed: some error"))
+
+  checkExpectedErrorHtmlState(mockErrorMessage)
+})
+
+test("handleUploadError function throws error and does not display error message if error HTML not present", () => {
+  setupNonErrorHtml()
+  const mockError = new Error("some error")
+  const mockErrorMessageElement: HTMLParagraphElement | null = document.querySelector(
+    ".errorMessage"
+  )
+
+  expect(() => {
+    handleUploadError(mockError)
+  }).toThrowError(new Error("Upload failed: some error"))
+})
+
+function checkExpectedErrorHtmlState(expectedRenderedErrorMessage: string) {
+  const mockFormElement: HTMLFormElement | null = document.querySelector(
+    "#file-upload-form"
+  )
+  const mockErrorElement: HTMLDivElement | null = document.querySelector(
+    ".govuk-error-summary"
+  )
+  const mockErrorMessageElement: HTMLParagraphElement | null = document.querySelector(
+    ".errorMessage"
+  )
+
+  if (mockFormElement) {
```

Comment:
Conditionals in test assertions are quite dangerous - this test would pass even if the elements weren't present. It also makes it hard to read the tests because the reader has to work out if it _should_ be true or false for any given test.

I don't know the best way to do this in Typescript, but the assertions should expect the elements to be there.

Reply:
Added a comment into the file to this affect.

Reply:
They'll fail for the right reason now, which is good! Hopefully we'll learn more about how to write good Jest tests as we go. 🙂